tests/checkers/unittest_non_ascii_name.py

```diff
@@ -0,0 +1,204 @@
+            pytest.param(
+                """
+                łol = "test" #@
+                """,
+                "Variable",
+                id="global_assign",
+            ),
+            pytest.param(
+                """
+                def foobar():
+                    łol="test"  #@
+                """,
+                "Variable",
+                id="function_variable",
+            ),
+            pytest.param(
+                """
+                for łol in os.listdir("."):  #@
+                    ...
+                """,
+                "Variable",
+                id="for_loop_variable",
+            ),
+            pytest.param(
+                """
+                [łoł
+                    for łol in os.listdir(".")  #@
+                ]
+                """,
+                "Variable",
+                id="inline_for_loop_variable",
+            ),
+        ],
+    )
+    def test_assignname(
+        self,
+        code: str,
+        assign_type: str,
+    ):
+        """Variables defined no matter where, should be checked for non ascii"""
+        assign_node = astroid.extract_node(code)
+
+        if not isinstance(assign_node, nodes.AssignName):
+            # For some elements we can't directly extract the assign
+            # node, so we have to manually look in the children for it
+            for child in assign_node.get_children():
+                if isinstance(child, nodes.AssignName):
+                    assign_node = child
+                    break
+
+        # Just to make sure we found the correct node
+        assert isinstance(assign_node, nodes.AssignName)
+
+        with self.assertAddsMessages(
+            pylint.testutils.MessageTest(
+                msg_id=self.EXPECTED_MSG,
+                node=assign_node,
+                args=(assign_type, "łol"),
+                confidence=pylint.interfaces.HIGH,
+            ),
+            ignore_position=True,
+        ):
+            self.checker.visit_assignname(assign_node)
+
+    def test_class_definition(self):
+        node = astroid.extract_node(
+            """
+            class FooBär: #@
+                ...
+            """
+        )
+        assert isinstance(node, nodes.ClassDef)
+
+        with self.assertAddsMessages(
+            pylint.testutils.MessageTest(
+                msg_id=self.EXPECTED_MSG,
+                node=node,
+                args=("Class", "FooBär"),
+                confidence=pylint.interfaces.HIGH,
+                line=2,
+                col_offset=0,
+                # pylint: disable-next=fixme
+                # TODO: Maybe we have to select something different
+                #       here, as line 3 is not that thing we expect
+                end_line=3,
+                end_col_offset=7,
+            )
+        ):
+            self.checker.visit_classdef(node)
+
+    def test_import_ignore_star(self):
+        """Special case from xyz import *, where '*' is not A-Za-z0-9 but still valid"""
```

Comment:
What happen in the import contain special unicode ? I suppose it's a false positive as the naming of dependencies it's not controlled by the user.

Say:
```python
from shadok import cołlander
```

Reply:
Then still the error is valid IMHO. Because it still would be an object with an invalid name within our namespace. That could still lead to problems.
The user always could use `from shadok import cołlander as collander`
Or simply put a pylint disable there.

Reply:
I like the alias solution. Maybe we could have a different message for it so it's clearer that this is what should be done ? `use-alias-for-non-ascii-import` ?

Reply:
@DanielNoord, this one :)

Reply:
Hmm yeah this is difficult. The importer can't really do much about this. Perhaps we could add something about aliasing to the description of the message? An additional message might indeed be overkill.

Reply:
Let's do that, there was nothing actionable in the warning and it's fairly small right now so it's a good solution I think.